server/src/main/java/org/opensearch/identity/tokens/AuthToken.java

```diff
@@ -8,11 +8,21 @@
 
 package org.opensearch.identity.tokens;
 
+import org.opensearch.common.io.stream.NamedWriteable;
+
 /**
  * Interface for all token formats to support to authenticate user such as UserName/Password tokens, Access tokens, and more.
  *
  * @opensearch.experimental
  */
-public interface AuthToken {
+public abstract class AuthToken implements NamedWriteable {
+
+    public TokenType getTokenType() {
+        return TokenType.NONE;
+    }
+
+    public String getTokenValue() {
```

Comment:
> what are the constraints on how this should work

I was thinking around the lines of should this value be cryptographically protected, or can it be in plain text?

There are sometimes when we need to be access the data in memory and we need a handle of the string that was presented that includes the password, but converting from in memory to something we serialize seems like it should have clear usage associated with it such as expiration, encryption.

I think it makes sense to turn a token into something that is serialized, but I think we might want to be very careful when this is done and it might be worthwhile to even wrap these tokens in another secure envelope to ensure they aren't tampered or readable outside the expected scopes.